src/main/java/pl/psnc/dei/service/BatchService.java

```diff
@@ -307,6 +324,80 @@ private String extractIdentifier(String key) {
 		return currentProject.getDatasets();
 	}
 
+	/**
+	 * Gets all of the records for specified Europeana dataset (by dataset id)
+	 * and upload them to the local database
+	 * @param request which contains processing settings
+	 * @return set of records for whole Europeana dataset (except excluded records or duplications)
+	 * @throws IllegalArgumentException when missing required params
+	 * @throws NotFoundException when cannot found data in local database
+	 */
+	public Set<Record> uploadDataset(UploadDatasetRequest request)
+			throws IllegalArgumentException, NotFoundException {
+		String projectName = request.getProjectName();
+		String datasetId = request.getDatasetId();
+		if (projectName == null || datasetId == null) {
+			throw new IllegalArgumentException();
+		}
+		Set<String> recordsIds = europeanaSearchService.getAllDatasetRecords(datasetId);
+		Set<String> excludedFiltered = filterExcluded(recordsIds, request.getExcludedRecords());
+		Set<String> numberLimited;
+		if (request.getLimit() > 0) {
+			int recordsToRetrieve = Math.min(excludedFiltered.size(), request.getLimit());
+			numberLimited = excludedFiltered.stream()
+					.limit(recordsToRetrieve)
```

Comment:
I would use this Math.min statement inside limit method instead of instatiating a variable for this

Reply:
moved `Math.min` statement